src/libs/actions/Report.js

```diff
@@ -1543,6 +1464,112 @@ function renameReport(reportID, reportName) {
+        }
+    }
+
+    const notificationPreference = lodashGet(allReports, [reportID, 'notificationPreference'], CONST.REPORT.NOTIFICATION_PREFERENCE.ALWAYS);
+    if (!ActiveClientManager.isClientTheLeader()) {
+        Log.info('[LOCAL_NOTIFICATION] Skipping notification because this client is not the leader');
+        return;
+    }
+
+    // We don't want to send a local notification if the user preference is daily or mute
+    if (notificationPreference === CONST.REPORT.NOTIFICATION_PREFERENCE.MUTE || notificationPreference === CONST.REPORT.NOTIFICATION_PREFERENCE.DAILY) {
+        Log.info(`[LOCAL_NOTIFICATION] No notification because user preference is to be notified: ${notificationPreference}`);
+        return;
+    }
+
+    // If this comment is from the current user we don't want to parrot whatever they wrote back to them.
+    if (isFromCurrentUser) {
+        Log.info('[LOCAL_NOTIFICATION] No notification because comment is from the currently logged in user');
+        return;
+    }
+
+    // If we are currently viewing this report do not show a notification.
+    if (reportID === lastViewedReportID && Visibility.isVisible()) {
+        Log.info('[LOCAL_NOTIFICATION] No notification because it was a comment for the current report');
+        return;
+    }
+
+    // If the comment came from Concierge let's not show a notification since we already show one for expensify.com
+    if (lodashGet(action, 'actorEmail') === CONST.EMAIL.CONCIERGE) {
+        return;
+    }
+
+    // When a new message comes in, if the New marker is not already set (newMarkerSequenceNumber === 0), set the marker above the incoming message.
+    const report = lodashGet(allReports, 'reportID', {});
+    if (lodashGet(report, 'newMarkerSequenceNumber', 0) === 0 && report.unreadActionCount > 0) {
+        const oldestUnreadSeq = (report.maxSequenceNumber - report.unreadActionCount) + 1;
+        setNewMarkerPosition(reportID, oldestUnreadSeq);
+    }
+
+    Log.info('[LOCAL_NOTIFICATION] Creating notification');
+    LocalNotification.showCommentNotification({
+        reportAction: action,
+        onClick: () => {
+            // Navigate to this report onClick
+            Navigation.navigate(ROUTES.getReportRoute(reportID));
+        },
+    });
+}
+
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
+    initWithStoredValues: false,
+    callback: (actions, key) => {
+        // reportID can be derived from the Onyx key
+        const reportID = parseInt(key.split('_')[1], 10);
+        if (!reportID) {
+            return;
+        }
+
+        _.each(actions, (action) => {
+            if (!action.timestamp) {
+                return;
+            }
+
+            // If we are past the deadline to notify for this comment don't do it
+            if (moment.utc(action.timestamp * 1000).isBefore(moment.utc().subtract(10, 'seconds'))) {
+                return;
+            }
```

Comment:
This is the only part I'm unsure about and left a question here in Slack...

https://expensify.slack.com/archives/C02HWMSMZEC/p1654549057704269

This is my replacement solution for `hasAlreadyBeenNotified: true` which I am not sure if we really need.